nntrainer/src/layer.cpp

```diff
@@ -215,4 +216,20 @@ void Layer::ensureName() {
   }
 }
 
+void Layer::print(std::ostream &out) const {
+  /// @todo change type enum to actual string value
+  out << '<' << typeid(*this).name() << "Object at" << this << '>' << std::endl
+      << "Layer Type: " << type << std::endl
+      << "Activation type: " << activation_type << std::endl
+      << "bias_initiate_to_zero: " << bias_init_zero << std::endl
+      << "Weight Decay: " << static_cast<int>(weight_decay.type) << " | "
+      << weight_decay.lambda << std::endl
+      << "Weight Initiation Type: " << weight_ini_type << std::endl
+      << "Loss: " << loss << std::endl
```

Comment:
IMO, loss and cost is redundant with each layer as it is same for the whole network.

Reply:
I'll delete cost 
loss is a bit subtle. Current implementation save's weight decay loss inside each layer (not in loss layer).

Reply:
may be change name to `Loss added`  and support when weight_decay is available?

Reply:
How about represent it as `weight regularization loss`?
For loss layer, it can just be `loss`.

Reply:
Sounds good, I'll do that